Feat: support selecting photos from tags

ahal opened the pull request:
Personally I'd prefer to use tags to select which photos show up on my frame, rather than adding photos to an album.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Per-account tag filtering and a new tag-based asset collection; tag collections appear alongside existing categories.
  * New "Show Tags Description" toggle to display comma-separated tag names in asset views, overlays, and client settings.

* **Documentation**
  * Configuration guide updated with tag setup, hierarchical/path matching examples, and required API permission.

* **Tests**
  * Added comprehensive unit tests plus minor test scaffolding improvements covering loading, pagination, matching, enrichment, and deduplication.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ahal commented on docs/docs/getting-started/configuration.md:
```diff
@@ -130,16 +132,21 @@ Accounts:
       - UUID
+    # UUID of Tags
+    Tags:  # string[]
+      - UUID
```
Maybe? I guess there could be issues when renaming tags. Mostly was trying to stay consistent with people and albums. If we used value for tags, then why not use a value for people and albums as well?

But I'm happy to change it to value if you prefer

ahal replied:
Re: easiest way, I didn't see a way to get it outside the API. But maybe I didn't look hard enough. Personally I'm fine with using the API, but you're right that for most users, that would be a barrier.

Probably worth mentioning that tags aren't enabled by default, so they're already power user territory.

JW-CH replied:
> Maybe? I guess there could be issues when renaming tags. Mostly was trying to stay consistent with people and albums. If we used value for tags, then why not use a value for people and albums as well?
> 
> But I'm happy to change it to value if you prefer

People and Albums could have the same `value` but not be the same; when I checked, that was not the case for `tags`. I just saw that the `tags` in Immich does not display a UUID; that's why I asked.

With `albums` or `people`, users can easily get the UUID via the URI.

I'd suggest using the tags `value` and, in the backend, getting all tags and their `ID`s manually (https://api.immich.app/endpoints/tags/getAllTags)

Please let me know what you think about this. :) I know that it would be easier to use the UUID for you, but for the end user it's better to have the value, since there won't be tag duplicates (afaik)

JW-CH reviewed (commented):
Could you run `make api` in the console to generate the `swagger.json`. Make sure the app is running and do it in another console :) also add the topmost line again. If you don't know what I want I can also do it myself :P

JW-CH commented on ImmichFrame.Core/Logic/Pool/TagAssetsPool.cs:
```diff
@@ -0,0 +1,73 @@
+                {
+                    Page = page,
+                    Size = batchSize,
+                    TagIds = [tagId],
```
the search can take multiple tags, why looping over and just take one?

JW-CH replied:
Just saw that the person pool does not do this too, but you could :)

ahal replied:
Given https://github.com/immichFrame/ImmichFrame/pull/537#discussion_r2692862080 I think I now actually do need to loop over them one by one. Otherwise there would be no way to know which tag to display as we aren't calling `GetAssetInfoAsync` anymore.

I'll publish this change soon and you can decide which one you prefer.

JW-CH commented on ImmichFrame.Core/Logic/Pool/TagAssetsPool.cs:
```diff
@@ -0,0 +1,73 @@
+                // Fetch full asset details to get tag information
+                foreach (var asset in tagInfo.Assets.Items)
+                {
+                    if (seenIds.Contains(asset.Id))
```
Also tracking `seenIds` is not needed when using one search (would be my guess :P)

ahal replied:
I'll remove this if we end up going with the approach that keeps `GetAssetInfoAsync`.

ahal commented on ImmichFrame.Core/Logic/Pool/TagAssetsPool.cs:
```diff
@@ -0,0 +1,73 @@
+                        continue;
+                    }
+
+                    if (asset.Tags == null)
```
I think this is actually a bug in Immich:
https://github.com/immich-app/immich/discussions/17361

Even though we're filtering by tag id, the tags aren't being returned in the response of `SearchAssetsAsync`, so another call is needed to get them.

But I think this was written back when I was using tag ids in the configuraiton. Now that we're specifying the tag value directly, we don't really need to get all the tags and we could just display that. The caveat is that in this current implementation, all tags for an asset are displayed, but if we drop the call to `GetAssetInfoAsync`, we'd only display the configured tag.

Personally I think this is fine and makes sense, so I'll go ahead and make that change. If you disagree I can revert it back.

ahal commented:
Thanks for the great review! I've addressed the comments I could, and guessed what you'd prefer to do about `SearchAssetsAsync` not returning tag info. Lmk which option you prefer.

Personally I like this because calling `GetAssetInfoAsync` per asset is way more expensive than calling `SearchAssetsAsync` per tag.

ahal commented on ImmichFrame.Core/Logic/Pool/TagAssetsPool.cs:
```diff
@@ -0,0 +1,69 @@
+        var tagAssets = new List<AssetResponseDto>();
+        var allTags = await apiCache.GetOrAddAsync($"allTags_{accountSettings.ImmichServerUrl}",
+            () => immichApi.GetAllTagsAsync(ct));
+        var tagValueToTag = allTags.ToDictionary(t => t.Value);
```
Oops, I need to update the documentation.

I realized that in Immich tags are case sensitive (e.g `People` and `people` are two separate tags). So we need to be case sensitive too.

JW-CH commented on ImmichFrame.Core/Logic/Pool/TagAssetsPool.cs:
```diff
@@ -0,0 +1,69 @@
+                {
+                    if (seenIds.Contains(asset.Id))
+                    {
+                        continue;
```
```suggestion
                        tagAssets.First(a => a.Id == asset.Id).Tags.Add(tag);
                        continue;
```

Something like this is needed to actually display all the tags on an item.

ahal commented:
> In general, I find the Tests quite big and confusing

Agreed, apologies for that. As you guessed I'm using Claude to help out. I think this next iteration is a lot more clear though. I also added an assertion that covers your first comment.
